### PR TITLE
Get suitable ppx changes into master

### PR DESCRIPTION
**Pull request by gaborigloi**
We want to port all ppx changes that we can to the master branch, to minimize the difference between the two branches to make the future merging simpler.

This PR renames a file and adds some brackets where necessary, these changes were required to make it compile with the newer 4.02.3 compiler. In addition, this PR also updates the version of the `_oasis` file to 0.4.

### CODE REVIEW

**Comment by gaborigloi** [comment]
This PR minimizes the differences from the PPX branch that is based on these changes: https://github.com/gaborigloi/xenopsd/tree/ppx_trunk

File: libvirt/xenops_server_libvirt.ml [hunk 1]
```diff
@@ -149,7 +149,7 @@ module Domain = struct
 			empty ~attr:["bridge", bridge] "source" output;
 			empty ~attr:["address", vif.Vif.mac] "mac" output;
 			(* TODO: the 'bridge' type doesn't like a script *)
-			(* empty ~attr:["path", !Path.vif_script] "script" output; *)
+			(* empty ~attr:["path", !Xpath.vif_script] "script" output; *)
```
Comment (mseri) [inline libvirt/xenops_server_libvirt.ml]: You fixed also the comments! Well done :+1:

File: _oasis [hunk 2]
```diff
@@ -56,7 +56,7 @@ Library xenopsd
     Xenops_server_plugin,
     Xenops_server_skeleton,
     Scheduler,
-    Path,
+    Xpath,
```
Comment (robhoes) [inline _oasis]: Can we pick are more meaningful name than `Xpath`?

File: libvirt/xenops_server_libvirt.ml (hunk 1 above)
Comment (gaborigloi) [inline libvirt/xenops_server_libvirt.ml]: `sed` did it :smile:

File: _oasis (hunk 2 above)
Comment (gaborigloi) [inline _oasis]: Hmm, it seems two other components also have similar modules:
```
xc/xc_path.ml
xl/xl_path.ml
```
It's unfortunate that more and more modules have to be renamed as OCaml gains new ones.

I'm not sure what this module is responsible for. Maybe `Paths` or `Resources` or `Resource`? Although then we would also have to rename the corresponding modules in `xc/` and `xl/`.

Comment (robhoes) [inline _oasis]: The module contains paths for resources in dom0. The `Xc_path` and `Xl_path` have paths specific to those backends. I'd be happy with any of the alternative names you suggested.

Comment (gaborigloi) [inline _oasis]: Ok, then I'll go for `Resources` unless someone has a different preference.

Comment (gaborigloi) [inline _oasis]: Actually, now I think `Dom0_resources` would be a more descriptive name.

Wouldn't renaming the module be a breaking change though, as I think `xenopsd` is exposed as a findlib library?

Comment (robhoes) [inline _oasis]: I can't think of anything else that may be using it. I think that the lib is just used by the multiple backends inside xenopsd.git.

Comment (gaborigloi) [inline _oasis]: Thanks for the info, updated the names of the modules. I'm going to double-check in planex-buildenv whether this breaks the other components.

**Comment by lindig** [comment]
This looks good to me. I didn't know that the parenthesis in `{ (f x) with ... }` became necessary.

**Comment by gaborigloi** [comment]
And the planex-buildenv build has passed, so apparently the module rename didn't break anything.

**Pull request merged**